io/include/pcl/io/grabber.h

```diff
@@ -266,6 +266,7 @@ namespace pcl
       const auto [iterator, success] =
     #else
       typename decltype(signals_)::const_iterator iterator;
+      std::string signame{typeid (T).name ()};
```

Comment:
This has to be moved further up, otherwise it is only defined in the `#else` case